src/init/features/project.ts

```diff
@@ -0,0 +1,149 @@
+import * as clc from "cli-color";
+import * as _ from "lodash";
+
+import * as Config from "../../config";
+import * as FirebaseError from "../../error";
+import { FirebaseProject, getProject, listProjects } from "../../firebaseApi";
+import * as logger from "../../logger";
+import { promptOnce, Question } from "../../prompt";
+import * as utils from "../../utils";
+
+const NO_PROJECT = "[don't setup a default project]";
+const NEW_PROJECT = "[create a new project]";
+
+/**
+ * Used in init flows to keep information about the project - basically
+ * a shorter version of {@link FirebaseProject} with some additional fields.
+ */
+export interface ProjectInfo {
+  id: string; // maps to FirebaseProject.projectId
+  label?: string;
+  instance?: string; // maps to FirebaseProject.resources.realtimeDatabaseInstance
+  location?: string; // maps to FirebaseProject.resources.locationId
+}
+
+/**
+ * Get the user's desired project, prompting if necessary.
+ * @returns A {@link ProjectInfo} object.
+ */
+async function getProjectInfo(options: any): Promise<ProjectInfo> {
+  if (options.project) {
+    return selectProjectFromOptions(options);
+  }
+  return selectProjectFromList(options);
+}
+
+/**
+ * Selects project when --project is passed in.
+ * @param options Command line options.
+ * @returns A {@link FirebaseProject} object.
+ */
+async function selectProjectFromOptions(options: any): Promise<ProjectInfo> {
+  let project: FirebaseProject;
+  try {
+    project = await getProject(options.project);
+  } catch (e) {
+    throw new FirebaseError(`Error getting project ${options.project}: ${e}`);
+  }
+  const projectId = project.projectId;
+  const name = project.displayName;
+  return {
+    id: projectId,
+    label: `${projectId} (${name})`,
+    instance: _.get(project, "resources.realtimeDatabaseInstance"),
+  };
+}
+
+/**
+ * Presents user with list of projects to choose from and gets project
+ * information for chosen project.
+ * @param options Command line options.
+ * @returns A {@link FirebaseProject} object.
+ */
+async function selectProjectFromList(options: any): Promise<ProjectInfo> {
+  let project: FirebaseProject | undefined;
```

Comment:
I think this declaration can be moved down to the `project = projects.find` line below - `project` isn't used elsewhere...